src/main.rs

```diff
@@ -147,7 +137,7 @@ fn compute_span_of_ram_sections(ram_entry: MemoryEntry, object: object::File) ->
     let mut found_a_section = false;
     for section in object.sections() {
         if let SectionFlags::Elf { sh_flags } = section.flags() {
-            if sh_flags & elf::SHF_ALLOC as u64 != 0 {
+            if (sh_flags & elf::SHF_ALLOC as u64) != 0 {
```

Comment:
Is it bit comparison here?

Reply:
Yes this is nitty-gritty 🗡️ !

Reply:
We should, but I don't know. `SHF_ALLOC` says that "Section occupies memory during execution.".

Reply:
I'd prefer not to block this PR on this, but figure out what the check does and add a comment in a separate PR.